src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final Integer DATE_INDEX_IN_RECORD = 0;
+    private static final Integer NAME_INDEX_IN_RECORD = 1;
+    private static final Integer HOURS_INDEX_IN_RECORD = 2;
+    private static final Integer PAYMENT_INDEX_IN_RECORD = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate localDateTo = LocalDate.parse(dateTo, FORMATTER);
+        LocalDate localDateFromRecord;
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ");
+        builder.append(dateFrom).append(" - ");
+        builder.append(dateTo);
+        for (String name : names) {
+            int salary = 0;
+            builder.append(System.lineSeparator()).append(name).append(" - ");
+            for (String record : data) {
+                String[] recordArray = record.split(" ");
+                localDateFromRecord = LocalDate.parse(recordArray[DATE_INDEX_IN_RECORD], FORMATTER);
+                if (name.equalsIgnoreCase(recordArray[NAME_INDEX_IN_RECORD])) {
+                    if ((localDateFromRecord.isAfter(localDateFrom)
+                            || localDateFromRecord.isEqual(localDateFrom))
+                            && (localDateFromRecord.isBefore(localDateTo)
+                            || localDateFromRecord.isEqual(localDateTo))) {
+                        salary += Integer.parseInt(recordArray[HOURS_INDEX_IN_RECORD])
+                                * Integer.parseInt(recordArray[PAYMENT_INDEX_IN_RECORD]);
```

Comment:
Lets divide logic for separate private method for make better readability cause this method is too long